pex/common.py

```diff
@@ -630,6 +630,8 @@ def get_parent_dir(path):
 
             def maybe_write_parent_dirs(path):
                 # type: (str) -> None
+                if path == strip_prefix:
+                    return
```

Comment:
What's the context for this change?

Reply:
Good catch actually. I missed an item on my cleanup checklist. This gets rid of a `..` entry in the `--layout packed` zips currently created by Pex. In Python 3.12 zipfile, these are now illegal. I need to change the cache format so that new Pex under 3.12 does not see the old zips since it will blow up if it does.

Reply:
Ok, I needed to refresh myself. Without this (the old behavior) you get packed wheel zips like:
```
Archive:  /home/jsirois/.pex/installed_wheel_zips/858d7ad606d58d8b13b4ac842dd378ff37e94387abc4b79e281576af39c64a49/py_spy-0.3.8-py2.py3-none-manylinux_2_5_x86_64.manylinux1_x86_64.whl
Zip file size: 3126344 bytes, number of entries: 10
drwxr-xr-x  2.0 unx        0 b- defN 80-Jan-01 00:00 ../
-rw-r--r--  2.0 unx       75 b- defN 80-Jan-01 00:00 .layout.json
drwxr-xr-x  2.0 unx        0 b- defN 80-Jan-01 00:00 .prefix/
drwxr-xr-x  2.0 unx        0 b- defN 80-Jan-01 00:00 .prefix/bin/
-rwxr-xr-x  2.0 unx  9838248 b- defN 80-Jan-01 00:00 .prefix/bin/py-spy
drwxr-xr-x  2.0 unx        0 b- defN 80-Jan-01 00:00 py_spy-0.3.8.dist-info/
-rw-r--r--  2.0 unx        4 b- defN 80-Jan-01 00:00 py_spy-0.3.8.dist-info/INSTALLER
-rw-r--r--  2.0 unx    16821 b- defN 80-Jan-01 00:00 py_spy-0.3.8.dist-info/METADATA
-rw-r--r--  2.0 unx        0 b- defN 80-Jan-01 00:00 py_spy-0.3.8.dist-info/REQUESTED
-rw-r--r--  2.0 unx      123 b- defN 80-Jan-01 00:00 py_spy-0.3.8.dist-info/WHEEL
10 files, 9855271 bytes uncompressed, 3125164 bytes compressed:  68.3%
```

The issue there is the 1st `../` entry. When Python 3.12 `zipfile` sees that, it says:
```
        if not arcname:
>           raise ValueError("Empty filename.")
E           ValueError: Empty filename.

../../../.pyenv/versions/3.12.0b3/lib/python3.12/zipfile/__init__.py:1761: ValueError
```

More directly:
+ 3.11
  ```
  $ mkdir /tmp/py311-unzip && python3.11 -mzipfile -e /home/jsirois/.pex/installed_wheel_zips/858d7ad606d58d8b13b4ac842dd378ff37e94387abc4b79e281576af39c64a49/py_spy-0.3.8-py2.py3-none-manylinux_2_5_x86_64.manylinux1_x86_64.whl /tmp/py311-unzip
  ```
+ 3.12
  ```
  $ mkdir /tmp/py312-unzip && python3.12 -mzipfile -e /home/jsirois/.pex/installed_wheel_zips/858d7ad606d58d8b13b4ac842dd378ff37e94387abc4b79e281576af39c64a49/py_spy-0.3.8-py2.py3-none-manylinux_2_5_x86_64.manylinux1_x86_64.whl /tmp/py312-unzip
  <frozen runpy>:128: RuntimeWarning: 'zipfile.__main__' found in sys.modules after import of package 'zipfile', but prior to execution of 'zipfile.__main__'; this may result in unpredictable behaviour
  Traceback (most recent call last):
    File "<frozen runpy>", line 198, in _run_module_as_main
    File "<frozen runpy>", line 88, in _run_code
    File "/home/jsirois/.pyenv/versions/3.12.0b3/lib/python3.12/zipfile/__main__.py", line 77, in <module>
      main()
    File "/home/jsirois/.pyenv/versions/3.12.0b3/lib/python3.12/zipfile/__main__.py", line 44, in main
      zf.extractall(curdir)
    File "/home/jsirois/.pyenv/versions/3.12.0b3/lib/python3.12/zipfile/__init__.py", line 1720, in extractall
      self._extract_member(zipinfo, path, pwd)
    File "/home/jsirois/.pyenv/versions/3.12.0b3/lib/python3.12/zipfile/__init__.py", line 1761, in _extract_member
      raise ValueError("Empty filename.")
  ValueError: Empty filename.
  ```

Reply:
I'll try to whip up a unit test for the new pex/common.py guard / behavior on top of the cache re-structure avoidance fix.